crates/api/src/kitsune.rs

```diff
@@ -0,0 +1,55 @@
+//! Kitsune2 top-level module related types.
+
+use crate::*;
+use std::sync::Arc;
+
+/// Handler for events coming out of Kitsune2.
+pub trait KitsuneHandler: 'static + Send + Sync + std::fmt::Debug {
+    /// Gather preflight data to send to a new opening connection.
+    /// Returning an Err result will close this connection.
+    fn preflight_gather(&self, peer_url: String) -> K2Result<bytes::Bytes>;
```

Comment:
What do you think of including `incoming`/`outgoing` in these function names, e.g. `preflight_outgoing_gather`?

Reply:
Done, and also gave these two default implementations: https://github.com/holochain/kitsune2/pull/40/commits/08d12d04e982d04c9211fb0a12827f440cc8b019